v8 #یوسف_قبادی

TGIR0 opened the pull request.

TGIR0 committed:
Add Support Android TV and Fix Endpoint Display

TGIR0 committed:
Add Android TV Support And Fix Endpoint Display

TGIR0 committed:
Fix Build buildTun2SocksAar

TGIR0 reviewed (commented):
this is fixed in buildTun2SocksAar commit

TGIR0 commented:
درود
مشکل نمایش بخش اندپوینت در اندروید تی وی حل شده و پشتیبانی هایی از اندروید تی وی اعمال شده
اندروید مورد نیاز به 7 تغییر یافت چون در اندروید 6 و پایین تر مشکل کرش هست که به علت تداخل با ویژگی های جدید است 
TargetSDK نیز به 36 ارتقا یافت 
بزودی مشکلاتی که برای کرش در اندروید های جدید هستن که به سیستم تم بیشتر ربط دارد حل خواهند شد

markpash commented:
> درود
> مشکل نمایش بخش اندپوینت در اندروید تی وی حل شده و پشتیبانی هایی از اندروید تی وی اعمال شده
> اندروید مورد نیاز به 7 تغییر یافت چون در اندروید 6 و پایین تر مشکل کرش هست که به علت تداخل با ویژگی های جدید است 
> TargetSDK نیز به 36 ارتقا یافت 
> بزودی مشکلاتی که برای کرش در اندروید های جدید هستن که به سیستم تم بیشتر ربط دارد حل خواهند شد 
> 

بسیار عالی، این رو به زودی بیلد میکنم و تست میکنم. خیلی ممنون ❤️

TGIR0 commented:
یه مشکل هست
قضیه اینه که کتابخانه هایی که  Arr استفاده میشن دیگه سمت اونا توسعه داده نمیشن 
و این باعث این شده با نسخه های جدیدتر Go همگام نباشه
و این باعث عدم پشتیبانی از سیستم عامل های جدیده 
باید ARR از اول با پکیج های بهتر جایگزین شه و ساختارش تغییر پیدا کنه 
@markpash @lewis-nixon

TGIR0 commented:
یکی از مشکلات فعلی اینه که Gradle بروز نمیشه کرد چون بایعث تداخل میشه و این باعث عدم پشتیبانی از SDK 36 و SDK 36.1 شه
Go هم با uTLS تداخل داره 
<img width="1387" height="976" alt="image" src="https://github.com/user-attachments/assets/681c2e50-d74f-43ff-8d4a-7416691de7b9" />

مشکل تم هم به کامپوننت های فعلی مربوط میشه و طبق قوانین جدید سیاست گوگل بهتره از Material UI 3 استفاده شه
همینطور که در https://github.com/bepass-org/oblivion/actions هست حتی در نسخه ای که تغییرات من اعمال نشده بیلد AAR مشکل داره
با تغییر نسخه های پکیج و چندین کانفیگ حل میشه ولی سیستم اتصال فیلترشکن از کار میوفته کاملا
به نظرتون چه کاری انجم بدیم بهتره

markpash commented:
> یه مشکل هست
> قضیه اینه که کتابخانه هایی که  Arr استفاده میشن دیگه سمت اونا توسعه داده نمیشن 
> و این باعث این شده با نسخه های جدیدتر Go همگام نباشه
> و این باعث عدم پشتیبانی از سیستم عامل های جدیده 
> باید ARR از اول با پکیج های بهتر جایگزین شه و ساختارش تغییر پیدا کنه 
> @markpash @lewis-nixon 

سعی میکنم که قسمت Go رو در روزهای آینده آپدیت کنم.

TGIR0 commented:
hello again
I Refactoring the project to Material 3 
when I finished will send as Commit

TGIR0 committed:
Merge branch 'main' into main

TGIR0 committed:
Upgrade Project To Material UI 3 (Beta) + Add Android 16 Support + Fix Lint and Version Errors

TGIR0 committed:
Merge branch 'main' of https://github.com/TGIR0/oblivion

TGIR0 committed:
Fix Tun2Socks Build

TGIR0 committed:
Upgrade Library + Fix Some Bugs + Test MASQUE

TGIR0 commented:
کل پکیج ها به اخرین نسخه بروز شدند
JAVA حداقل به 17 و از جاوای 25 نیز پشتیبانی میکند
با اخرین ساختار GRADLE بازسازی شد کانفیگ ها
بهبود ها و هشدار ها رفع شد 
کتابخانه های موجود به اخرین نسخه رسمی بروز شدند
در حال تست MASQUE و بازطراحی دکمه اتصال هستم
دکمه اتصال از یک پکیج دیگر استفاده میکند که منسوخ و دارای اختلال است و نیازمند همگام سازی ان با طراحی متریال میباشد
در دیباگ بیلد خطایی یا هشداری مشاهده نشده 
مشکل بیلد نشدن TUN با فعالسازی CGO حل شد
بزودی GEO IP و GEO Site نیز اضافه خواهم کرد
اندروید 16 بطور کامل پشتیبانی میشود

markpash commented:
> کل پکیج ها به اخرین نسخه بروز شدند
> JAVA حداقل به 17 و از جاوای 25 نیز پشتیبانی میکند
> با اخرین ساختار GRADLE بازسازی شد کانفیگ ها
> بهبود ها و هشدار ها رفع شد 
> کتابخانه های موجود به اخرین نسخه رسمی بروز شدند
> در حال تست MASQUE و بازطراحی دکمه اتصال هستم
> دکمه اتصال از یک پکیج دیگر استفاده میکند که منسوخ و دارای اختلال است و نیازمند همگام سازی ان با طراحی متریال میباشد
> در دیباگ بیلد خطایی یا هشداری مشاهده نشده 
> مشکل بیلد نشدن TUN با فعالسازی CGO حل شد
> بزودی GEO IP و GEO Site نیز اضافه خواهم کرد
> اندروید 16 بطور کامل پشتیبانی میشود
> 

سمت warp plus یه آپدیت ریز دادم که باعث میشه بیلد بشه روی گو ۱.۲۴, میتونی همین رو بدون اضافه کردن قابلیت جدید، آماده مرج کنی تا بتونم یه تست بگیرم؟

TGIR0 commented:
> > کل پکیج ها به اخرین نسخه بروز شدند
> > JAVA حداقل به 17 و از جاوای 25 نیز پشتیبانی میکند
> > با اخرین ساختار GRADLE بازسازی شد کانفیگ ها
> > بهبود ها و هشدار ها رفع شد
> > کتابخانه های موجود به اخرین نسخه رسمی بروز شدند
> > در حال تست MASQUE و بازطراحی دکمه اتصال هستم
> > دکمه اتصال از یک پکیج دیگر استفاده میکند که منسوخ و دارای اختلال است و نیازمند همگام سازی ان با طراحی متریال میباشد
> > در دیباگ بیلد خطایی یا هشداری مشاهده نشده
> > مشکل بیلد نشدن TUN با فعالسازی CGO حل شد
> > بزودی GEO IP و GEO Site نیز اضافه خواهم کرد
> > اندروید 16 بطور کامل پشتیبانی میشود
> 
> سمت warp plus یه آپدیت ریز دادم که باعث میشه بیلد بشه روی گو ۱.۲۴, میتونی همین رو بدون اضافه کردن قابلیت جدید، آماده مرج کنی تا بتونم یه تست بگیرم؟

چشم یه کامیت جدا میدم براش

TGIR0 commented:
یه مشکلی هست سمت برنامه که به go 1.24 همگام نیست یکم بهت مهلت بده این مشکل رو حل کنم و از اخرین نسخه warp plus نیز استفاده کنم و نتیجه رو بفرستم

TGIR0 committed:
Repair Project Codes

TGIR0 committed:
Repair Project

TGIR0 commented:
با آخرین استاندارد کل مشکلات موجود در لاگ ها و مشکلات شناسایی شده
بخش های غیرفعال
منسوخ
اضافی
همگی اعمال شدند
سایفون برای رفع تداخل بین نسخه ها و مشکل بیلد حذف گردید
مقادیر ترجمه ها بهبود یافته اند

TGIR0 committed:
VPN Service Upgrade

TGIR0 commented:
@markpash در وارپ پلاس کل سایفون را حذف کن با TLS تداخل داره و همینطور کتابخانه های golang.org/x/crypto 
به همین علت الان نمیتونم خروجی بگیرم 
در سمت برنامه حذف میکنم تو هم در سمت هسته این کار را کن

markpash commented:
> @markpash در وارپ پلاس کل سایفون را حذف کن با TLS تداخل داره و همینطور کتابخانه های golang.org/x/crypto 
> به همین علت الان نمیتونم خروجی بگیرم 
> در سمت برنامه حذف میکنم تو هم در سمت هسته این کار را کن
> 

فکر کنم یه راه بهتری برای این باشه، بذار امروز یه تست بگیرم

TGIR0 committed:
Fix TLS + Remove psiphon + Fix Some Leaks and Enhnce in UI + UX

TGIR0 commented:
مشکل اینه که سایفون دیگه با کریپتو سازگار نیست و همگام نمیشه 
کریپتو و سایفون هم زمان توی پروژه تداخل ایجاد میکنن 
سیستم TLS هم یه گلیچ داره که بی نهایت اجرا میشه و هی تکرار میشه با بروزرسانی و ساده سازی و تغییر ساختار میشه حل کرد

TGIR0 committed:
update Git files

TGIR0 committed:
Delete tun2socks/build.ps1

TGIR0 committed:
Delete tun2socks_build_debug.log

TGIR0 committed:
Delete tun2socks/setup-env.ps1

TGIR0 committed:
Delete tun2socks/go_updates.txt

TGIR0 commented:
همه چیز حدودا سالم به نظر میرسه جز طراحی یه دکمه که دارم روش کار میکنم

TGIR0 commented:
امیدوارم از قضد بنده آزرده خاطر نشوید
در حال پایدار سازی نهایی هستم و یک فایل نصبی نیز برای شما ارسال میکنم تا در گوشی تست کنید و نتیجه را به من بگویید
وارپ پلاس رو کاستومایز کردم و شاید با یه کتابخانه تغییر بدم چون دیگه قابلیت مقابله با فیلترینگ رو نداره
بهبود های ظاهری و چند ویژگی نیز اضافه میکنم 
البته با اجازه از شما

TGIR0 committed:
Final Release (Stable)

TGIR0 committed:
Fixed

TGIR0 commented:
https://github.com/bepass-org/warp-plus/pull/304

TGIR0 commented:
[Oblivion.zip](https://github.com/user-attachments/files/23655965/Oblivion.zip)
نسخه بیلد شده
نگران نباشید امن است میتوانید بررسی کنید

TGIR0 committed:
V8.0.1

TGIR0 committed:
v.8.1

TGIR0 committed:
Update warp-plus

TGIR0 committed:
fix Bugs

TGIR0 commented:
درود
هسته ساختارش بهتر شد 
نسخه های کتابخانه ها بروز شد 
با go 1.25.4 تداخل ندارد
سایفون حذف گردید
سیستم یافتن آیپی سالم بهبود یافت
نشتی اطلاعات و نواقض از بین رفت
سیستم تست خوکار قبل از بیلد و سیستم لاگ اضافه گردید
برنامه بهبود های جزئی اعمال شد که برخی عبارتند از
بهبود ساختار پشتیبانی از اندروید ۱۶
ارتقای کتابخانه ها به اخرین نسخه
تغییر نسخه جاوا از ۱۷ به ۲۵
ارتقای سیستم لاگ و بهبود نمایش وضعیت اتصال
افزودن تم مشکی برای دستگاه های Oled
یکپارچه سازی طراحی و تم
استاندارد سازی رنگ تم ها
تغییر مجوز های فیلترشکن با قوانین اندروید ۱۴ به بالا
ارتقا و سازگاری با Gradle 9 
کاملسازی و رفع مشکلات موجود از نسخه های پیشین مانند:
ایکون لودینگ نا متناسب
پشتیبانی نکردن از Android TV
همگام نبودن با زبان پیشفرض در اولین اجرا
بسته نشدن صفحه غیرفعالسازی بهینه سازی باتری
مشکلات همگام نبودن اعلان موجود با وضعیت فعلی برنامه
مشکل کارکرد دکمه موجود در منوی سریع
مصرف منابع بسیار بالا و نامتعادل
ساختار نادرست اجرا و بستن برنامه
سیستم پس زمینه مشکل‌ساز
بهینه نبودن کد ها و بیلد سخت و کند
پشتیبانی نکردن از ARM
Min SDK اشتباه
عدم پشتیبانی از اندروید های بالاتر 
ریسپانسیو نبودن 

یک لیست چنجلاگ کامل از نسخه ۷ به ۸ میدم بزودی

TGIR0 commented:
> [Oblivion.zip](https://github.com/user-attachments/files/23655965/Oblivion.zip) نسخه بیلد شده نگران نباشید امن است میتوانید بررسی کنید

برای آخرین نسخه منتظر بیلد باشید
ساختار کارکرد بیلد توسط گیتهاب نیز بهبود یافته است

TGIR0 commented:
@markpash

TGIR0 committed:
Merge branch 'main' into main

TGIR0 committed:
Disable MASQUE

TGIR0 committed:
Merge branch 'main' of https://github.com/TGIR0/oblivion

TGIR0 commented:
هنوز MASQUE برای اندروید توسعه ندادن
وقتی که MASQUE برای اندروید منتشر شد فعال میکنم موقتا در حالت بزودی قرار داده شد
msg="failed to run warp: MASQUE implementation is currently being integrated. Please check back in the next update or use Standard Warp."

TGIR0 committed:
optimize for google play store policy

TGIR0 commented:
خسته نباشید
یه مدت مجبور شدم روی بقیه پروژه ها کار کنم
یکیش تکمیل شد
یکیش وضعیتش خوبه
اون تموم شد بر میگردم به تغییر هسته این برنامه و از vwarp استفاده میکنم به جای هسته فعلی

alijamshidiyan1-tech commented:
سلام کار نمیکنه ؟ تنظیمات رو چطور‌وارد کنم ؟

TGIR0 commented:
خسته نباشید 
بعد از مدت ها قطعی برگشتم با تغییر اکوسیستم
190 فایل تغییر یافتن
ساختار برنامه بازطراحی میشه برای هسته جدید 
دفعه قبلی هسته براش در نظر گرفته نشده بود ولی اینبار از vwarp به طور موقت و کاندوئیت سایفون استفاده میکنم که پرچمدار های فعلی هستن

TGIR0 commented:
> سلام کار نمیکنه ؟ تنظیمات رو چطور‌وارد کنم ؟

هسته برنامه غیرفعاله 
صرفا در اول رفع مشکلات کرش و باگ و اجرا و بهینه سازی و پشتیبانی بروز در اولویت بود در نسخه 8

TGIR0 commented:
@markpash 
این تغییرات حجیم که درحال تکمیل هستن کل ساختار جاوا رو حذف میکنه و به کاتولین ترجمه میشه 
Gradle 9 با اندروید 16 به طور پیشرفته تر 
بهینه سازی فوق العاده زیاد (نزدیک 80 درصد نسبت به نسخه 8 و بهبود چندین برابری نسبت به نسخه 7)
vwarp یک فورک از وارپ پلاسه که توسط کاربر بنام وید توسعه داده میشه برای شرایط ایران 
لایه های ایمنی بالا اضافه میکنه
از این هسته به جا vwarp استفاده میکنیم که برای این کاربر هم مفید باشه که بازخورد بگیره و بهبود بده ساختارشو

TGIR0 commented:
<img width="471" height="886" alt="image" src="https://github.com/user-attachments/assets/ae7c3415-1263-4bb8-a2c2-90ce00ccdd28" />

Commit:
v8.0_Pre

TGIR0 commented:
درود
همانطور که با خبر هستید قطعی 65 روزه و بیشتر اینترنت در ایران وجود دارد
گزارشاتی که عرض میکنم قبل از خاموشی اتفاق افتادن:
1-همزمان با توسعه سمت برنامه oblivion شروع به بازسازی چندین هسته های معروف مانند Singbox به زبان برنامه نویسی RUST کردم
همانطور که با خبر هستید به خاطر نشتی هسته Xray یوتوب متوجه ایپی های ایرانی شد
زبان Rust به علت ماهیت ذاتی اش اجازه کرش و Leak نمیده 
در امنیت هم لایه های بسیار بهتری از زبان Go داره
هسته برنامه در فاز اول فقط به کد های Rust ترجمه شده اند
اگر قطعی وجود نداشت اکنون تا حد کافی تست های ممکن روی هسته انجام شده بود و اماده اعمال روی هسته ممکن بود
در سمت برنامه:
پشتیبانی از اندروید 17 
ارتقا کتابخانه ها مانند MUI
ارتقا kotlin به نسخه پایدار و ارتقا رسمی و پایدار کل پروژه به JAVA 25 
تغییر کل قایل های Java به kt برای بهینه سازی اکوسیستم 
اعمال بهینه سازی و رفع مشکلات heap و مموری مربوط به نسخه های اخیر JAVA که از سوی سازنده رفع شده
اعمال فلگ های بهینه سازی جاوا بر روی کتابخانه های مربوطه
تغییر اکوسیستم Gradle به ساختار جدید واقع در مستندات Gradle و kotlin 
در زیر گوشه چشمی از تغییرات رو مشاهده میکنید:

## 2026-02-19

### Upgraded
- Kotlin: `2.2.10` → `2.3.20-RC` (`gradle/libs.versions.toml`)
- KSP: `2.3.5` → `2.3.6` (`gradle/libs.versions.toml`)
- Kotlin compiler args: `-Xjvm-default=all` (deprecated) → `-jvm-default=enable` (`build.gradle`, `app/build.gradle`)
- Added Kotlin EAP repository (restricted to `org.jetbrains.kotlin*`) to support Kotlin `2.3.20-RC` artifacts (`settings.gradle`)

### Notes
- Removed the gradle-versions-plugin (`dependencyUpdates`) because it crashes with `java.util.ConcurrentModificationException` on this Gradle/AGP stack. Use `version_audit.ps1` instead.
- `detekt-gradle-plugin` currently triggers a Gradle 9 deprecation warning (`ReportingExtension.file(String)`), so it must be updated before moving to Gradle 10.

## 2026-04-25 (initial changes)

### Upgraded

#### Core Toolchain
- Kotlin: `2.3.20-RC` → `2.4.0` (later reverted, see final corrections)
- KSP: `2.3.6` → `2.4.0-1.0.31` (later aligned with Kotlin downgrade, see final corrections)
- AGP: `8.14.0` → `9.1.0` (`gradle/libs.versions.toml`)
- Gradle: `9.2.1` → `9.3.1` (wrapper + `gradle-wrapper.properties`)
- JDK: `21` → `25` (toolchain JVM, daemon JVM, `gradle.properties`)
- NDK: `29.0.14206865` (unchanged, but verified on AGP 9)

#### AndroidX & Compose
- Compose BOM: `2026.02.00` → `2026.04.00`
- Navigation Compose: `2.9.7` → `2.10.0-alpha01`
- Activity Compose: `1.13.0-alpha01` → `1.13.0-alpha02`
- Core KTX: `1.18.0-rc01` → `1.18.0`
- Lifecycle: `2.10.0` (unchanged)
- AppCompat: `1.7.1` → `1.7.2`
- Material Design: `1.14.0-alpha09` → `1.14.0-alpha10`
- ConstraintLayout: `2.2.1` (unchanged)
- Fragment: `1.8.9` (unchanged)
- Core Splashscreen: `1.2.0` (unchanged)
- ProfileInstaller: `1.4.1` (unchanged)

#### Plugin Dependencies
- Google Services: `4.4.4` → `4.4.5`
- Firebase Plugins: Crashlytics `3.0.6` → `3.0.7`; Perf `2.0.2` → `2.0.3`
- Detekt: `1.23.8` → `1.24.0`
- Spotless: `8.2.1` → `8.3.0`
- Dependency Analysis: `3.5.1` → `3.6.0`

#### Core Libraries
- Hilt: `2.59.2` (unchanged)
- OkHttp: `5.3.2` (unchanged)
- Coroutines: `1.10.2` (unchanged)
- MMKV: `2.3.0` (unchanged here; later bumped, see final corrections)
- Coil: `3.3.0` → `3.4.0-alpha01`
- LeakCanary: `3.0-alpha-8` → `3.0-alpha-9`

### Massive Java to Kotlin Migration
- Converted all remaining Java source files to Kotlin, including:
  - `ApplicationLoader` (now fully Hilt-aware, with proper error isolation and Timber logging)
  - `BaseActivity`, `StateAwareBaseActivity`
  - All activities: `MainActivity`, `SettingsActivity`, `LogActivity`, `SplashScreenActivity`, `SplitTunnelActivity`, `InfoActivity`
  - All adapters: `BypassListAppsAdapter`, `EndpointsBottomSheet`, `SplitTunnelOptionsAdapter`
  - Utility classes: `FileManager`, `SystemUtils`, `ColorUtils`, `LocaleManager`, `HostPortParser`, etc.
  - Custom views: `Icon`, `TouchAwareSwitch`
  - DNS and network modules: `NetworkModule`, `DnsUriParserTest`, `DnsExecutionPlannerTest`, etc.
  - Build-time classes: `FileExistsValueSource`, `OptionalPropertiesValueSource`
  - Removed all Java sources; project is now fully Kotlin-based.
- Updated `app/build.gradle` to use `alias(libs...)` exclusively for plugins and dependencies.
- Removed deprecated `vectorDrawables.useSupportLibrary` and redundant `buildConfig` flag from `app/build.gradle`.
- Migrated release signing configuration from hardcoded values to external `keystore.properties` (gitignored), with a `validateReleaseSigning` task.
- Rewrote gomobile tasks (`buildTun2socksAar`) using `providers.exec()` for Configuration Cache compatibility.
- Switched `FileManager.initialize` call from `BaseActivity` to `ApplicationLoader` to avoid redundant initialisation.
- Refactored `FileManager` into a singleton `object`, added `Keys` constants, and replaced manual locking with `ReentrantReadWriteLock`.
- Centralised all VPN configuration building in `FileManager.getVpnConfig()` under a single read-lock for consistency.

### Lint & Code Quality
- Lint baseline (`lint-baseline.xml`) completely emptied after fixing all reported issues:
  - Updated dependency versions.
  - Removed unused resources (drawables, colors, strings, dimensions/styles, arrays).
  - Fixed icon density issues and removed redundant `png` copies in favour of `anydpi` vector drawables.
  - Replaced `LinearLayout` in `toast.xml` with a single `TextView` using compound drawables.
  - Removed overdraw on root elements by relying on theme backgrounds.
  - Hardcoded endpoint text moved to string resources.
- Upgraded `detekt.yml` to the latest official baseline, updated rule names and properties.
- **Deleted the lint baseline file entirely** (zero lint warnings now).
- Removed `gradle-versions-plugin` because it crashes on this Gradle/AGP stack.
  Use `version_audit.ps1` instead.

### ProGuard / R8
- Rewrote `proguard-rules.pro` with modern best practices:
  - Enabled line number retention and source file renaming for Crashlytics visibility.
  - Added `-assumenosideeffects` to strip `android.util.Log` calls in release builds.
  - Retained native library rules for `tun2socks` and `go`.

### Build Infrastructure
- Updated `gradle.properties`:
  - JVM args: `-Xmx2048M` (from 1536M); removed obsolete `-XX:MaxPermSize`.
  - Enabled parallel builds (`org.gradle.parallel=true`).
  - Set `org.gradle.warning.mode=all`.
  - Added `android.builtInKotlin=true` for AGP 9.x (though it is the default, explicit is safer).
  - `android.enableR8.fullMode=true` kept for clarity (default since AGP 8).
- Updated wrapper scripts (`gradlew`, `gradlew.bat`, `gradle-wrapper.properties`) to match the current Gradle wrapper version.
- Updated `devshell.nix` and `flake.nix` to target JDK 26 and Go 1.26 for future‑proofing (the project still compiles with JDK 25).
- Updated `version_audit.ps1` to compare against the **latest stable** Gradle version (not RCs), avoiding false positives.
- Revised `.gitignore` to track new build artifacts and Kotlin caches properly.

### Configuration & App Manifest
- Overhauled `AndroidManifest.xml`:
  - Added `android:foregroundServiceType="dataSync|specialUse"` and the corresponding `PROPERTY_SPECIAL_USE_FGS_SUBTYPE` to the VPN service (required since Android 14).
  - Removed the dangerous `QUERY_ALL_PACKAGES` permission; kept only safe `<queries>` entries.
  - Ensured compatibility with Predictive Back gesture on Android 16+.
  - Removed deprecated `usesCleartextTraffic` attribute; network security is now controlled exclusively via `networkSecurityConfig`.
- Redesigned `dns_providers.json` into a structured catalog (`pinned`, `verified`, `community`) with Iranian providers, `selectionWeight` fields, and health‑check rules for the custom DNS engine.
- Resolved Git conflict in `dimens.xml`, keeping all spacing and radius tokens with proper documentation.

### Core Tunnel (Go) — tun2socks.go
- Complete rewrite of the tunnel core for safety and resource hygiene:
  - Eliminated global mutable state; all lifecycle is now managed by a `Tunnel` struct.
  - Added `Start`/`Wait`/`Stop` non‑blocking semantics with proper goroutine draining.
  - Added panic recovery in the main worker goroutine with full stack‑trace logging.
  - Capturing stdout/stderr is done with `os.Pipe`; both are restored cleanly on shutdown.
  - Used `sync.Pool` for zero‑allocation log pipeline buffers.
  - Input validation (`validateOptions`) reports all errors at once using `errors.Join`.
  - All file descriptors and goroutines are guaranteed to be released before `Stop` returns.

## 2026-04-25 (final corrections — post‑audit)

### Version Corrections
- **Kotlin:** `2.4.0` reverted to `2.3.21` — `2.4.0` is not yet released (planned June‑July 2026).
- **KSP:** `2.4.0-1.0.31` → `2.3.21-1.0.31` (aligned with Kotlin downgrade).
- **Gradle:** `9.3.1` → `9.4.1` — latest stable as of late April 2026.
- **MMKV:** `2.3.0` → `2.4.0` (released March 2026).
- **Coil:** `3.4.0-alpha01` → `3.4.0` (stable release).
- **Navigation Compose:** `2.10.0-alpha01` → `2.9.7` (reverted to stable; the alpha is not recommended for production).
- **Spotless:** `8.3.0` → `8.4.0` (latest release).
- **Go (devshell):** `1.25` → `1.26.1` (latest stable; aligned with `devshell.nix`).
- **JDK (devshell):** `25` → `26` (aligned with `devshell.nix`; the project’s compilation toolchain remains on JDK 25 until full validation).

### Config Adjustments
- Added `android.builtInKotlin=true` to `gradle.properties` (explicit for AGP 9.x).
- Updated `Gradle_Playbook.md` and `devshell.nix` to reflect final version numbers.
- Finalised `tun2socks.go` with all reviewed improvements (race‑condition fix, panic recovery, stdout/stderr capture hygiene, resource‑pool cleanup).
- Updated `flake.nix` to pin NDK 29.0.14206865 (matching the project) and JDK 26.

### Notes
- Kotlin `2.4.0` should be adopted when it reaches stable (target: mid‑2026).
- Gradle `9.4.1` officially supports JDK 26, but the project’s compilation toolchain remains on JDK 25 until all dependencies are verified against JDK 26.
- The dependency analysis plugin (`com.autonomousapps.dependency-analysis`) stays at `3.6.0` — no newer stable version available.
- All Java files have been deleted and the project is now **100% Kotlin** (including `buildSrc` and tests). No Java source remains.

Commit:
Merge branch 'main' of https://github.com/TGIR0/oblivion

Commit:
Remove All Java Files

Convert .Java to .kt

Commit:
Remove Gradle old files (Stage 1)

remove old grade files and moved to new files

Commit:
Refactor Package Lists

use only used packages and upgrade to latest version + delete some log files + update md files and other files

TGIR0 commented:
هنوز کار داره تا تکمیل کنم یکم منتظر باشید
هسته رو برای اخر نگه داشتم که برنامه مشکلی نداشته باشه 
چون حجم تغییرات بسیار زیاده برای شما و من دردسر هست که فایل ها بررسی شه 
بعد از تکمیل فایل نهایی بیلد شده رو تست کنید در صورت مطلوب بودن به سمت هسته برویم

Commit:
New Gradle Files use new gradle and kotlin structure

refractor the gradle and kotlin to latest state of gradle/kotlin docs and app

TGIR0 committed:
Update go.yml

TGIR0 committed:
disable GithubBot

TGIR0 committed:
Rename android.yml to android.yml.disable

TGIR0 committed:
Rename release.yaml to release.yaml.disable

Commit:
upgrade SDK to 37 (Android 17) + upgrade libs

Commit:
upgrade UI Libs from MUI 2 to MUI3 and new backend

Commit:
Update settings.gradle.kts

Commit:
restructure project packages and update version catalog

Commit:
Complete UI refactoring to Material UI 3

Commit:
Update build.gradle.kts

Commit:
Update .gitignore

Commit:
Update .gitignore

Commit:
Update ThemeHelper.kt

Commit:
Complete UI refactoring to Material UI 3

Commit:
Remove Core for Planed Refactor

Commit:
fix

Commit:
remove trash

Commit:
Remove Trash files

Commit:
Update README.md

Commit:
Report Changes

Commit:
Upgrade for New Gradle Features

Commit:
Update gradle.properties

Commit:
Fix Dependency Checker

Commit:
Use Gradle 9.6.0 Features

Commit:
Update build.gradle.kts

Commit:
Add support Android 17 + Remove Tun2socks Temporarily

Commit:
Update AndroidManifest.xml

Commit:
Update libs.versions.toml

Commit:
Update gradle-wrapper.properties

Commit:
Update .gitignore

Commit:
Update build.gradle.kts

Commit:
Update libs.versions.toml

Commit:
Update README.md

Commit:
Update gradle.properties

Commit:
Gradle Upgrade

Commit:
Refactor UI To use v7 Layout + Colors and Upgrade Dependency and fix some bugs

Commit:
Upgrade UI + improve security

Commit:
start testing some changes for Github support Latest Build

Commit:
UI Refactoring

Commit:
Update detekt.yml

Commit:
Update libs.versions.toml

Commit:
UI Refactoring

Commit:
Create .envrc

Commit:
Fix Battery Optimization Bugs and improve Log and error showing dialogs

Commit:
Add log security Features and Test Files

Commit:
Add UI Test Files

Commit:
Update AndroidManifest.xml

Commit:
Update go.mod

Commit:
Delete tun2socks.go

Commit:
Update HomeScreen.kt

Commit:
Update UiTestTags.kt

Commit:
Update APP Side to support Hev Socks5 TUN and New Log + Better Error Handle

Commit:
Improve App Security

Commit:
Create VpnServiceStartPolicy.kt

Commit:
Update QuickStartService.kt

Commit:
add and import Test and Handle Features

Commit:
Update Github workflows to support New core (Core is coming Soon)

Commit:
Update Gradle Files to support New Core

Commit:
Update OptionalPropertiesValueSource.kt

Commit:
Update FileExistsValueSource.kt

Commit:
Add Hev TUN Service

Commit:
initialize DNS Support (Not functional now)

Commit:
Add WG Protocol Support App Side

Commit:
Update TunnelConfigV2.kt

Commit:
Create DnsUserProviderRepository.kt

Commit:
Update CountryCodeExtensions.kt

Commit:
update Files

rhoggs-bot-test-account committed:
Update README.md

TGIR0 committed:
Revise README for Oblivion v8 release

Updated README to reflect changes in version 8, including new features and system requirements.

rhoggs-bot-test-account committed:
Disable temporarily

rhoggs-bot-test-account committed:
Update AndroidManifest.xml

rhoggs-bot-test-account committed:
add Aether Core Translation

rhoggs-bot-test-account committed:
Update build.gradle.kts

rhoggs-bot-test-account committed:
Update update_deps.ps1

rhoggs-bot-test-account committed:
Update version_audit.ps1

rhoggs-bot-test-account committed:
Delete keystore.jks

rhoggs-bot-test-account committed:
Update go.mod

rhoggs-bot-test-account committed:
Update libs.versions.toml

rhoggs-bot-test-account committed:
Update verification-metadata.xml

rhoggs-bot-test-account committed:
Update SplitTunnelMode.kt

rhoggs-bot-test-account committed:
Merge branch 'main' of https://github.com/TGIR0/oblivion

rhoggs-bot-test-account committed:
Update gradle.lockfile

rhoggs-bot-test-account committed:
remove trash

rhoggs-bot-test-account committed:
Update UPGRADE_NOTES.md

rhoggs-bot-test-account committed:
Update libs.versions.toml

rhoggs-bot-test-account committed:
Update gradle-wrapper.properties

rhoggs-bot-test-account committed:
Remove Dead Codes and old files

rhoggs-bot-test-account committed:
Update proguard-rules.pro

rhoggs-bot-test-account committed:
Update gradle.lockfile

TGIR0 commented:
گویا هسته اتر رو دارید وارد میکنید
برای همین دیگه PR لازم نیست 
جداگانه توسعه هسته Warp Plus ادامه میدم
به تازگی به دنبال احیا و بازنویسی Warp plus با Rust هستم که حداقل این برنامه ای که از 2025 توسعه پیدا میکنه و سمت برنامه تکمیل شده ای داره حداقل به یک نسخه موفق در تست های داخلی برسه و هر روزی که Aether از توسعه خارج شد یا متد فیلترینگ اماده نبود اماده باشه 
هسته قرار نیست تا مدت طولانی در گیتهاب رو نمایی شه و تست داخلی برای اینترنت ایران و افزودن سایفون به طور قوی برای از بین بردن حساسیت سایت ها به ایپی کلادفلر و افزایش پایداری و قابلیت استفاده مجزا در ویندوز آماده سازی بشه 
به امید آزادی
موفق باشید
